nerfstudio/model_components/losses.py

```diff
@@ -248,17 +250,18 @@ def urban_radiance_field_depth_loss(
         Depth loss scalar.
     """
     # Expected depth loss
-    expected_depth_loss = (termination_depth - predicted_depth) ** 2
+    expected_depth_loss = ((termination_depth - predicted_depth) * depth_mask) ** 2
 
     # Line of sight losses
     target_distribution = torch.distributions.normal.Normal(0.0, sigma / URF_SIGMA_SCALE_FACTOR)
     termination_depth = termination_depth[:, None]
     line_of_sight_loss_near_mask = torch.logical_and(
         steps <= termination_depth + sigma, steps >= termination_depth - sigma
     )
+    line_of_sight_loss_near_mask = torch.logical_and(line_of_sight_loss_near_mask, depth_mask[:, None])
     line_of_sight_loss_near = (weights - torch.exp(target_distribution.log_prob(steps - termination_depth))) ** 2
     line_of_sight_loss_near = (line_of_sight_loss_near_mask * line_of_sight_loss_near).sum(-2)
-    line_of_sight_loss_empty_mask = steps < termination_depth - sigma
+    line_of_sight_loss_empty_mask = torch.logical_and(steps < termination_depth - sigma, depth_mask[:, None])
```

Comment:
It is difficult for me to judge the correctness. Can you just mask the loss like in ds_nerf_loss?